src/decoration.c

```diff
@@ -1,27 +1,4213 @@
+    ewram_1f000.items = gSaveBlock1.playerRoomDecor;
+    ewram_1f000.pos = gSaveBlock1.playerRoomDecorPos;
+    ewram_1f000.size = sizeof gSaveBlock1.playerRoomDecor;
+    ewram_1f000.isPlayerRoom = 1;
+}
+
+void sub_80FE2B4(void)
+{
+    u16 palettes[3];
+    memcpy(palettes, gUnknown_083EC654, sizeof gUnknown_083EC654);
+    LoadPalette(&palettes[2], 0xdf, 2);
+    LoadPalette(&palettes[1], 0xd1, 2);
+    LoadPalette(&palettes[0], 0xd8, 2);
+}
+
+void Task_DecorationPCProcessMenuInput(u8 taskId)
+{
+    if (!gPaletteFade.active)
+    {
+        if (gMain.newKeys & DPAD_UP)
+        {
+            PlaySE(SE_SELECT);
+            gUnknown_020388D4 = MoveMenuCursor(-1);
+            sub_80FE394();
+        }
+        if (gMain.newKeys & DPAD_DOWN)
+        {
+            PlaySE(SE_SELECT);
+            gUnknown_020388D4 = MoveMenuCursor(1);
+            sub_80FE394();
+        }
+        if (gMain.newKeys & A_BUTTON)
+        {
+            PlaySE(SE_SELECT);
+            gUnknown_083EC604[gUnknown_020388D4].func(taskId);
+        } else if (gMain.newKeys & B_BUTTON)
+        {
+            PlaySE(SE_SELECT);
+            gpu_pal_decompress_alloc_tag_and_upload(taskId);
+        }
+    }
+}
+
+void sub_80FE394(void)
+{
+    MenuFillWindowRectWithBlankTile(2, 15, 27, 18);
+    MenuPrint(gUnknown_083EC624[gUnknown_020388D4], 2, 15);
+}
+
+void gpu_pal_decompress_alloc_tag_and_upload(u8 taskId)
+{
+    sub_8072DEC();
+    MenuZeroFillWindowRect(0, 0, 10, 9);
+    MenuFillWindowRectWithBlankTile(2, 15, 27, 18);
+    FreeSpritePaletteByTag(6);
+    if (ewram_1f000.isPlayerRoom == 0)
+    {
+        ScriptContext1_SetupScript(gUnknown_0815F399);
+        DestroyTask(taskId);
+    } else
+    {
+        ReshowPlayerPC(taskId);
+    }
+}
+
+void sub_80FE418(u8 taskId)
+{
+    sub_80FE5AC(taskId);
+}
+
+void sub_80FE428(u8 taskId)
+{
+    InitMenu(0, 1, 1, 4, gUnknown_020388D4, 9);
+    sub_80FE394();
+    gTasks[taskId].func = Task_DecorationPCProcessMenuInput;
+}
+
+void sub_80FE470(u8 decoCat, u8 a1, u8 a2, u8 palIdx)
+{
+    u8 *strptr;
+    u8 v0;
+    v0 = sub_8072CBC();
+    // PALETTE {palIdx}
+    strptr = gStringVar4;
+    strptr[0] = EXT_CTRL_CODE_BEGIN;
+    strptr[1] = 5;
+    strptr[2] = palIdx;
+    strptr += 3;
+    strptr = StringCopy(strptr, gUnknown_083EC5E4[decoCat]);
+    strptr = sub_8072C14(strptr, sub_8134194(decoCat), 0x56, 1);
+    *strptr++ = 0xba;
+    strptr = sub_8072C14(strptr, gDecorationInventories[decoCat].size, 0x68, 1);
+    strptr[0] = EXT_CTRL_CODE_BEGIN;
+    strptr[1] = 5;
+    strptr[2] = v0;
+    strptr[3] = EOS;
+    MenuPrint(gStringVar4, a1, a2);
+}
+
+void sub_80FE528(u8 taskId)
+{
+    u8 decoCat;
+    MenuDrawTextWindow(0, 0, 14, 19);
+    for (decoCat=0; decoCat<8; decoCat++)
+    {
+        if (ewram_1f000.isPlayerRoom == 1 && gTasks[taskId].data[11] == 0 && decoCat != DECOCAT_DOLL && decoCat != DECOCAT_CUSHION)
+        {
+            sub_80FE470(decoCat, 1, 2 * decoCat + 1, 13);
+        } else
```

Comment:
I saw that syntax elsewhere in the repo.  Refer back to comment on style/formatting guide.